helloworld.py

```diff
@@ -0,0 +1,141 @@
+
+from flask import Flask, request
+import uuid
+import json
+import milvus_model
+from pymilvus import MilvusClient, DataType, Collection
+
+import boto3
+import os
+import re
+
+from dotenv import load_dotenv
+
+from PIL import Image
+# from io import BytesIO
+import timm
+from sklearn.preprocessing import normalize
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+import torch
+
+from transformers import AutoFeatureExtractor, AutoModelForImageClassification
+
+class FeatureExtractor:
+    def __init__(self, modelname):
+        # Load the pre-trained model
+        self.model = timm.create_model(
+            modelname, pretrained=True, num_classes=0, global_pool="avg"
+        )
+        self.model.eval()
+
+        # Get the input size required by the model
+        self.input_size = self.model.default_cfg["input_size"]
+
+        config = resolve_data_config({}, model=modelname)
+        # Get the preprocessing function provided by TIMM for the model
+        self.preprocess = create_transform(**config)
+
+    def __call__(self, imagepath):
+        # Preprocess the input image
+        input_image = Image.open(imagepath).convert("RGB")  # Convert to RGB if needed
+        input_image = self.preprocess(input_image)
+
+        # Convert the image to a PyTorch tensor and add a batch dimension
+        input_tensor = input_image.unsqueeze(0)
+
+        # Perform inference
+        with torch.no_grad():
+            output = self.model(input_tensor)
+
+        # Extract the feature vector
+        feature_vector = output.squeeze().numpy()
+
+        return normalize(feature_vector.reshape(1, -1), norm="l2").flatten()
+
+client = MilvusClient(
+    uri=os.getenv("MILVUS_ENDPOINT")
+)
+
+endpoint_url = "http://" + os.getenv("BUCKET_HOST") + ":"+ os.getenv("BUCKET_PORT")
+
+s3 = boto3.client(
+    "s3",
+    aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
+    aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
+    endpoint_url=endpoint_url,
+    use_ssl=False,
+    verify=False,
+)
+# Store bucket name
+bucket_name = os.getenv("BUCKET_NAME")
+
+object_type = os.getenv("OBJECT_TYPE")
+
+
+i = 0
+
+app = Flask(__name__)
+
+@app.route('/', methods=['POST'])
+def pythonvectordbappceph():
+
+    collection_name = re.sub('-', '_', bucket_name)
+    app.logger.debug(request.data)
+    event_data = json.loads(request.data)
+    object_key = event_data['Records'][0]['s3']['object']['key']
+    app.logger.debug(object_key)
+    if not client.has_collection(collection_name=collection_name):
+        client.create_collection(
+            collection_name=collection_name,
+            dimension=os.getenv("VECTOR_DIMENSION"),
+            enable_dynamic_field=True,
+            metric_type="COSINE",
```

Comment:
should we use the same `metric_type` for all data cases?
maybe we should add that as another parameter to the pod?